src/connector.js

```diff
@@ -0,0 +1,612 @@
+                numberOfResults = numberOfResults.substring( 0, numberOfResults.length - 7 ) + ',' + numberOfResults.substring( numberOfResults.length - 7 );
+            }
+
+            querySummaryElement.innerHTML = ( ( querySummaryState.query != "" && !params.isAdvancedSearch ) ? querySummaryTemplateHTML : noQuerySummaryTemplateHTML )
+				.replace( '%[numberOfResults]', numberOfResults )
+				.replace( '%[query]', querySummaryState.query );
+		}
+	}
+	else if ( querySummaryState.hasError ) {
+		querySummaryElement.textContent = "";
+		querySummaryElement.innerHTML = resultErrorTemplateHTML;
+	}
+}
+
+// Update pagination
+function updatePagerState( newState ) {
+	pagerState = newState;
+	pagerElement.textContent = "";
+	let innerHTML = "";
+	if ( pagerState.hasPreviousPage ) {
+		const liNode = document.createElement( "li" );
+		liNode.innerHTML = previousPageTemplateHTML;
+		const aNode= liNode.querySelector( 'a' );
+		aNode.onclick = ( e ) => { 
+            pagerController.previousPage();
+            window.scrollTo( { top: 200, behavior: 'smooth' } );
+        };
+		pagerElement.appendChild( liNode );
+	}
+	pagerState.currentPages.forEach( ( page ) => {
+		const liNode = document.createElement( "li" );
+		if ( page == pagerState.currentPage ) {
+			liNode.setAttribute( "class","active" );
+		}
+		const pageNo = page;
+		liNode.innerHTML = pageTemplateHTML.replace( '%[page]', pageNo );
+		const aNode = liNode.querySelector( 'a' );
+		aNode.onclick = ( e ) => {
+            pagerController.selectPage( pageNo );
+            window.scrollTo( { top: 200, behavior: 'smooth' } );
+        };
+		pagerElement.appendChild( liNode );				
+	} );
+	if ( pagerState.hasNextPage ) {
+		const liNode = document.createElement( "li" );
+		liNode.innerHTML = nextPageTemplateHTML;
+		const aNode = liNode.querySelector( 'a' );
+		aNode.onclick = ( e ) => { 
+            pagerController.nextPage(); 
+            window.scrollTo( { top: 200, behavior: 'smooth' } );
+        };
+		pagerElement.appendChild( liNode );
+	}
+}
+
+// Subscribe to Headless controllers
+searchBoxController.subscribe( () => updateSearchBoxState( searchBoxController.state ) );
+resultListController.subscribe( () => updateResultListState( resultListController.state ) );
+querySummaryController.subscribe( () => updateQuerySummaryState( querySummaryController.state ) );
+pagerController.subscribe( () => updatePagerState( pagerController.state ) );
+
+// Listen to "Enter" key up event for search action
+if ( searchBoxElement ) {
+	searchBoxElement.onkeyup = ( e ) => {
+		lastCharKeyUp = e.keyCode;
+		if( e.keyCode === 13 ) {
+			e.preventDefault(); // Ensure it is only this code that runs
+			//searchBoxController.submit(); // not required, click on search button occurs automatically...
+		}
+		else {
+			searchBoxController.updateText( e.target.value );
+		}
+	};
+	searchBoxElement.onfocus = ( e ) => {
+		lastCharKeyUp = null;
+		searchBoxController.showSuggestions();
```

Comment:
Strange that here we do have a function call to show it and elsewhere we do directly hide it by modifying the element style. I recommend to use a function to hide that suggestion box.

Reply:
some adjustment in the reference implementation (Add more working example) might by required

Reply:
Please review code now, it should be acceptable to go ahead with what we have now